### DISCUSSION TITLE
Can I delete a page programmatically?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#1692] [Q&A] [answered]
### Question

Hi All,

I wanted to know if we can delete the page routes that we create using the ```@ui.page```. 

For ex: I was testing some menu buttons and so I did a simple page function like this:
```python
@ui.page('/{icon}')
def icon_page(icon):
    ui.label(f'This page is for {icon}')
```

This is just for me to test out if the button clicks were working and the pages were opening etc, because I was trying to insert my buttons in middle of a long code. 

Now that is done, is there a way to delete the pages? I mean even if I remove the functions, the pages are already created and I can manually type in the URL and still access them. I guess one way is to restart the app.

But is there a way to programmatically delete the pages from python?

Thanks,
Anindya

**Comment by rodja** [comment, marked as answer]
We use something like this

```py
app.routes[:] = [routes for routes in app.routes if routes.path != '/home']
```

Which will remove the `/home` page from FastAPI.

**Reply by Anindya088** [reply]
Hi @rodja , thanks so much, exactly what I was looking for 😄

**Reply by falkoschindler** [reply]
You can also use
```py
from nicegui import app

app.remove_route('/home')
```
